src/plugins/intel_cpu/src/transformations/snippets/x64/op/brgemm_cpu.cpp

```diff
@@ -38,30 +38,45 @@ BrgemmCPU::BrgemmCPU(const Output<Node>& A, const Output<Node>& B, const Output<
     constructor_validate_and_infer_types();
 }
 
+void BrgemmCPU::constructor_validate_and_infer_types() {
+    INTERNAL_OP_SCOPE(BrgemmCPU_constructor_validate_and_infer_types);
+    validate_inputs();
+
+    // During ctor call, BrgemmCPU doesn't know his port descriptors.
+    // So we use port descs from source inputs
```

Comment:
I'm not sure I fully understand this problem, let's discuss it offline.
I also don't understand why this became a problem now, what changed compared to the previous approach?

Reply:
Discussed and resolved offline, added layouts to constructor parameters. Thanks